src/main/java/io/vlingo/actors/Environment.java

```diff
@@ -69,6 +69,10 @@ void addChild(final Actor child) {
     children = children.plus(child);
   }
 
+  void removeChild(final Actor child) {
+    children = children.minus(child);
```

Comment:
I am concerned that this would cause a problem with simultaneous child removal. Is there another way to do a thread safe removal?

Reply:
Note that I think `children.plus(child)` is safe because an actor may only create a child in it's constructor or during message handling. However, removing a child is driven by the child's thread, unless removal would be based on a message send/delivery.

Reply:
`removeChild(Actor)` method is `synchronized` now.